nex/upgrade.go

```diff
@@ -125,6 +127,17 @@ func UpgradeNex(ctx context.Context, logger *slog.Logger, newVersion string) (st
 
 	shasum := hex.EncodeToString(h.Sum(nil))
 
+	err = os.Rename(nexPath+".new", nexPath)
```

Comment:
Mixed formatting, seems like not too big of a deal though, as long as the linter doesn't mind.